ta4j-core/src/main/java/org/ta4j/core/BaseTradingRecord.java

```diff
@@ -77,7 +79,7 @@
     /**
      * The recorded positions
      */
-    private List<Position> positions = new ArrayList<>();
+    private TreeMap<Integer, Position> sortedPositions = new TreeMap<>();
```

Comment:
Did not see that `getPositions(`) returns a list. Hence, it can remain

`TreeMap<Integer, Position> sortedPositions`

or 

`TreeMap<Integer, Position> allPositions`

Choose what you want. Does not matter.